common/config/persistence.go

```diff
@@ -77,20 +77,6 @@ func (c *Persistence) Validate() error {
 				ErrPersistenceConfig,
 				c.SecondaryVisibilityStore)
 		}
-		if isPrimaryEs && isSecondaryEs {
-			// ElasticSearch config for visibilityStore and secondaryVisibilityStore must be the same except for
-			// `indices.visibility` config key and private fields - this is a restriction due to global ES client
```

Comment:
If we opt for only this option I think we need a better gate here. I worry that we are removing a protection

Reply:
We don't really have a good way to gate this.